chaoscenter/web/src/components/ProjectDashboardCardContainer/ProjectDashboardCardContainer.tsx

```diff
@@ -0,0 +1,149 @@
+import { Button, ButtonVariation, Card, Container, Layout, Popover, Text } from '@harnessio/uicore';
+import React, { useState } from 'react';
+import { Color, FontVariation } from '@harnessio/design-system';
+import { Classes, PopoverInteractionKind, Position, Menu, Dialog } from '@blueprintjs/core';
+import { QueryObserverResult, RefetchOptions, RefetchQueryFilters } from '@tanstack/react-query';
+import { useHistory } from 'react-router-dom';
+import { ListProjectsOkResponse, Project } from '@api/auth';
+import CustomTagsPopover from '@components/CustomTagsPopover';
+import { useStrings } from '@strings';
+import ProjectDashboardCardMenuController from '@controllers/ProjectDashboardCardMenu';
+import { setUserDetails, toSentenceCase } from '@utils';
+import { useAppStore } from '@context';
+import css from './ProjectDashboardCardContainer.module.scss';
+
+interface ProjectDashboardCardProps {
+  projects: Project[] | undefined;
+  listProjectRefetch: <TPageData>(
+    options?: (RefetchOptions & RefetchQueryFilters<TPageData>) | undefined
+  ) => Promise<QueryObserverResult<ListProjectsOkResponse, unknown>>;
+}
+
+export default function ProjectDashboardCardContainer(props: ProjectDashboardCardProps): React.ReactElement {
+  const { projects, listProjectRefetch } = props;
+  const [projectId, setProjectId] = useState<string>();
+  const { getString } = useStrings();
+  const history = useHistory();
+  const { updateAppStore } = useAppStore();
+
+  const handleClick = (project: Project) => {
+    handleProjectSelect(project);
+  };
+
+  const handleProjectSelect = (project: Project): void => {
+    updateAppStore({ projectID: project.projectID, projectName: project.name });
+    setUserDetails({
+      projectID: project.projectID
+    });
+    history.push(`/`);
+  };
+
+  return (
+    <Container width="100%">
+      <Container padding={{ top: 'large', bottom: 'xxlarge' }} className={css.cardsMainContainer}>
+        {projects?.map(project => {
+          return (
+            <Card
+              onClick={() => handleClick(project)}
```

Comment:
We can directly call `handleProjectSelect` here.

Reply:
![image](https://github.com/litmuschaos/litmus/assets/92683836/d528043c-9ff4-41bb-a309-c3b5e7a45eea)


gives this error

can be solved by `onClick={() => handleProjectSelect}`

is it okay wdyt

Reply:
yeah, sure LGTM then.